### PR TITLE
Ports Barometers from Yogstation

### PR DESCRIPTION
**Pull request by KomradeSpectre**
:cl: Super3222, TheMythicGhost, DaedalusGame
add: Adds a barometer function to the standard atmos analyzer.
imageadd: Adds a new sprite for the atmos analyzer to resemble a barometer.
/:cl:

[why]: Despite the unpredictability of storms being a 'FEATURE', having some way of predicting future storms can assist miners to make decisions to tough out the storm, seek shelter, or return to base more intuitively than waiting for the storm warning and making a mad dash. This also can be extended to work with future weather types by adding barometer_predictable = TRUE to said weather type.

Port from https://github.com/yogstation13/yogstation/pull/1285 with some shoehorning.

### CODE REVIEW

**Comment by DaedalusGame** [comment]
Sprites are from Oracle (uraniummeltdown pr'd them before, but they got denied for gas meters)

**Comment by Repukan** [comment]
Miners get a good 10 second warning to seek shelter, they don't need a way to power game storms or it removes the only danger it had (unpredictability)

**Comment by WJohn** [comment]
No. We have analyzers that handle this already. Throw this function into those so they're slightly more useful and so these things aren't a one-trick object that will never be taken.

**Comment by KomradeSpectre** [comment]
If you can lead me to what item you're meaning that has the weather detecting functionality, by all means. I ported this because it was incredibly useful and immersive in Yog, and I thought it would be here.

**Comment by WJohn** [comment]
The air analyzer found in all mechanical toolboxes. The black and blue handheld scanner that everyone *immediately* throws the fuck away because it's effectively worthless.

**Comment by DaedalusGame** [comment]
it's worthless on lavaland because it scans atmosphere and all of lavaland has the same atmosphere

**Comment by Qustinnus** [comment]
what are you on about @DaedalusGame

**Comment by DaedalusGame** [comment]
you'll figure it out eventually, try reading the comment again

**Comment by DaedalusGame** [comment]
but they don't look like barometers

**Comment by ChuckTheSheep** [comment]
If you're going to be pedantic, why would spessmen use analogue barometers? Don't miners have an non-atmos analyzer already? (I think it might be for ore) if you care about the oldtimey look Just slap a pressure meter on that thing.

**Comment by optimumtact** [comment]
wjohn already neatly answered why this is not a great change.

I know it's more fun to add unique sprites, but existing stuff needs a lot lot more love

**Comment by KomradeSpectre** [comment]
I've changed the sprite for the atmos analyzer and removed my old barometer sprite. We'll have to wait for the IconDiff bot to regenerate the preview.

**Comment by Cyberboss** [comment]
It's broken atm, post it yourself please

**Comment by KomradeSpectre** [comment]
The different icon is as follows:
image

**Comment by nicbn** [comment]
Can you make it a different item instead of a mode please?

**Comment by DaedalusGame** [comment]
leave

**Comment by KomradeSpectre** [comment]
>Can you make it a different item instead of a mode please?

**AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA!!!!!!!**

**Comment by nicbn** [comment]
Analyzer already is pretty useful on its own for atmos techs. Miners should get a different tool.

**Comment by DaedalusGame** [comment]
go argue with oranges and wjohn

**Comment by nicbn** [comment]
No you go

**Comment by DaedalusGame** [comment]
i already did further up

**Comment by ChuckTheSheep** [comment]
Tie it into the ore scanner...

**Comment by DaedalusGame** [comment]
>shoehorn it into a different device

**Comment by ChuckTheSheep** [comment]
Still related to mining.

**Comment by KomradeSpectre** [comment]
I honestly feel this should be reviewed again as I have completed what was originally requested, and it works as intended.

**Comment by DaedalusGame** [comment]
Chuck, this also works in snowdin

**Comment by ChuckTheSheep** [comment]
:man_shrugging:

File: code/game/objects/items/devices/scanners.dm [hunk 1]
```diff
@@ -403,6 +406,69 @@ GAS ANALYZER
...
 
+/obj/item/device/analyzer/AltClick(mob/user) //Barometer output for measuring when the next storm happens
+	..()
+
+	if(user.canUseTopic(src))
+		var/turf/T = get_turf(user)
+		if(!T)
+			return
+
+		playsound(src, 'sound/effects/pop.ogg', 100)
```
Comment (optimumtact) [inline code/game/objects/items/devices/scanners.dm]: shouldn't you put this after the cooldown?

**Comment by KomradeSpectre** [comment]
Done as requested @optimumtact

**Pull request merged**